src/H5A.c

```diff
@@ -1057,15 +1057,15 @@ H5Aread(hid_t attr_id, hid_t dtype_id, void *buf /*out*/)
  *--------------------------------------------------------------------------*/
 herr_t
 H5Aread_async(const char *app_file, const char *app_func, unsigned app_line, hid_t attr_id, hid_t dtype_id,
-              void *buf, hid_t es_id)
+              void *buf /*out*/, hid_t es_id)
 {
     H5VL_object_t *vol_obj   = NULL;            /* Object for attr_id */
     void *         token     = NULL;            /* Request token for async operation        */
     void **        token_ptr = H5_REQUEST_NULL; /* Pointer to request token for async operation        */
     herr_t         ret_value = SUCCEED;         /* Return value */
 
     FUNC_ENTER_API(FAIL)
-    H5TRACE7("e", "*s*sIuii*xi", app_file, app_func, app_line, attr_id, dtype_id, buf, es_id);
```

Comment:
Looks like the *x -> x in this line and the change below might be unintentional?

Reply:
No, that's intentional and the result of the bin/trace script updates that parse the parameters better.

Reply:
This is probably just from my ignorance of bin/trace in general, but the only changes to bin/trace I see are adding cases for H5ES_event_complete_func_t and H5ES_event_insert_func_t. How does that affect needing the asterisk for the void * buf parameter here?

Reply:
> This is probably just from my ignorance of bin/trace in general, but the only changes to bin/trace I see are adding cases for H5ES_event_complete_func_t and H5ES_event_insert_func_t. How does that affect needing the asterisk for the void * buf parameter here?

Adding the "/*out*/" comment to the parameters indicates to the bin/trace script that the buffer contents are not valid at function entry and are therefore an 'x' (i.e. totally opaque to tracing) and not a '*x' (which indicates a pointer to a buffer of bytes).